src/stuf/stuf_zds/client.py

```diff
@@ -88,142 +82,136 @@ def xml_value(xml, xpath, namespaces=nsmap):
-                "zds_documenttype_omschrijving_inzending"
-            ],
-            "zds_zaakdoc_vertrouwelijkheid": self.options[
-                "zds_zaakdoc_vertrouwelijkheid"
-            ],
-            "referentienummer": str(uuid4()),
-            "global_config": self._global_config,
-        }
-
-    def _wrap_soap_envelope(self, xml_str: str) -> str:
-        return loader.render_to_string(
-            "stuf_zds/soap/includes/envelope.xml",
+    def build_base_context(self) -> dict[str, Any]:
+        context = super().build_base_context()
+        now = timezone.now()
+        context.update(
             {
-                "soap_version": self.service.soap_service.soap_version,
-                "soap_use_wss": (
-                    self.service.soap_service.endpoint_security
-                    in [EndpointSecurity.wss, EndpointSecurity.wss_basicauth]
+                "tijdstip_registratie": fmt_soap_datetime(now),
+                "datum_vandaag": fmt_soap_date(now),
+                "gemeentecode": self.options.get("gemeentecode", ""),
+                "zds_zaaktype_code": self.options.get("zds_zaaktype_code", ""),
+                "zds_zaaktype_omschrijving": self.options.get(
+                    "zds_zaaktype_omschrijving"
+                ),
+                "zds_zaaktype_status_code": self.options.get(
+                    "zds_zaaktype_status_code"
+                ),
+                "zds_zaaktype_status_omschrijving": self.options.get(
+                    "zds_zaaktype_status_omschrijving"
                 ),
-                "wss_username": self.service.soap_service.user,
-                "wss_password": self.service.soap_service.password,
-                "wss_created": fmt_soap_date(timezone.now()),
-                "wss_expires": fmt_soap_date(
-                    timezone.now() + timedelta(minutes=STUF_ZDS_EXPIRY_MINUTES)
+                "zaak_omschrijving": self.options.get("omschrijving", ""),
+                "zds_documenttype_omschrijving_inzending": self.options.get(
+                    "zds_documenttype_omschrijving_inzending", ""
                 ),
-                "content": mark_safe(xml_str),
-            },
+                "zds_zaakdoc_vertrouwelijkheid": self.options.get(
+                    "zds_zaakdoc_vertrouwelijkheid", ""
+                ),
+                "global_config": GlobalConfiguration.get_solo(),
+            }
         )
+        return context
 
     def _make_request(
         self,
+        soap_action: str,
         template_name: str,
         context: dict,
-        endpoint_type,
-        soap_action: str = "",
-    ) -> Tuple[Response, Element]:
-
-        request_body = loader.render_to_string(template_name, context)
-        request_data = self._wrap_soap_envelope(request_body)
+        endpoint_type: str,
+    ) -> tuple[Response, Element]:
+        # URL for logging purposes
+        _url = self.service.get_endpoint(type=endpoint_type)
```

Comment:
bit of duplication and a risk of code going out of sync... this is only used for logging purposes. I'd rather use a requests hook to process/log the response, but that also requires some state :/